src/handlers/annotate.ts

```diff
@@ -0,0 +1,157 @@
+    const response = await octokit.rest.issues.listComments({
+      owner: repository.owner.login,
+      repo: repository.name,
+      issue_number: context.payload.issue.number,
+      per_page: 100,
+    });
+    const commments = response.data;
+    if (commments.length > 1) {
+      const commentBeforeAnnotate = commments[commments.length - 2];
+      await commentChecker(context, commentBeforeAnnotate, scope);
+    } else {
+      logger.error("No comments before the annotate command");
+    }
+  } else {
+    const { data } = await octokit.rest.issues.getComment({
+      owner: repository.owner.login,
+      repo: repository.name,
+      comment_id: parseInt(commentId, 10),
+    });
+    await commentChecker(context, data, scope);
+  }
+}
+
+/**
+ * Checks if the comment is similar to a existing issue.
+ * If a similar issue is found, a footnote is added to the comment.
+ * @param context The context object
+ * @param comment The comment object
+ * @param scope The scope of the annotation
+ **/
+export async function commentChecker(context: Context, comment: Comment, scope: string) {
+  const {
+    logger,
+    adapters: { supabase },
+    payload,
+  } = context;
+  let commentBody = comment.body;
+  if (!commentBody) {
+    logger.info("Comment body is empty", { commentBody });
+    return;
+  }
+  commentBody = removeAnnotateFootnotes(commentBody);
+  const similarIssues = await supabase.issue.findSimilarIssues({
+    markdown: commentBody,
+    currentId: comment.node_id,
+    threshold: context.config.warningThreshold,
+  });
+  if (similarIssues && similarIssues.length > 0) {
+    let processedIssues = await processSimilarIssues(similarIssues, context, commentBody);
+    processedIssues = processedIssues.filter((issue) =>
+      filterByScope(scope, payload.repository.owner.login, issue.node.repository.owner.login, payload.repository.name, issue.node.repository.name)
+    );
+    if (processedIssues.length > 0) {
+      logger.info(`Similar issue which matches more than ${context.config.warningThreshold} already exists`, { processedIssues });
+      await handleSimilarIssuesComment(context, payload, commentBody, comment.id, processedIssues);
+      return;
+    }
+  }
+  context.logger.info("No similar issues found for comment", { commentBody });
+}
+
+function filterByScope(scope: string, repoOrg: string, similarIssueRepoOrg: string, repoName: string, similarIssueRepoName: string): boolean {
+  switch (scope) {
+    case "global":
+      return true;
+    case "org":
+      return repoOrg === similarIssueRepoOrg;
+    case "repo":
+      return repoOrg === similarIssueRepoOrg && repoName === similarIssueRepoName;
+    default:
+      return false;
+  }
+}
+
+async function handleSimilarIssuesComment(
+  context: Context,
+  payload: Context["payload"],
+  commentBody: string,
+  commentId: number,
+  issueList: IssueGraphqlResponse[]
+) {
+  // Find existing footnotes in the body
+  const footnoteRegex = /\[\^(\d+)\^\]/g;
```

Comment:
Maybe we should prefix the footnotes for issue de duplication with a special character to target them more precisely @shiv810 rfc